### PR TITLE
Fixes #4529 - foreman.url option and SRV

### PR DESCRIPTION
**Pull request by lzap**
Not tested yet, I am going to test after the discussion.

### CODE REVIEW

File: extra/ovirt-node-plugin-foreman/src/discover-host.rb.in [hunk 1]
```diff
@@ -74,46 +75,62 @@ def error msg
...
+  URI.parse(url)
+rescue
+  return nil
+end
+
+def discover_by_dns_srv
+  domain = `dnsdomainname`.chomp
+  resolver = Resolv::DNS.new
+  type = Resolv::DNS::Resource::IN::SRV
+  result = resolver.getresources("_foreman._tcp.#{domain}", type).first
```
Comment (domcleal) [inline extra/ovirt-node-plugin-foreman/src/discover-host.rb.in]: By the way, http://projects.puppetlabs.com/issues/3669#note-19

Comment (lzap) [inline extra/ovirt-node-plugin-foreman/src/discover-host.rb.in]: > By the way, http://projects.puppetlabs.com/issues/3669#note-19

Thanks, changing this to `x-foreman` :-D

## 

Later,

 Lukas "lzap" Zapletal
 irc: lzap #theforeman

**Comment by lzap** [comment]
Done with testing, the build is here: http://yum.theforeman.org/discovery/nightly/

@GregSutcliffe ^

**Comment by lzap** [comment]
Initiated one more image scratch build, found issue with SRV search. Wait 20 minutes before checking the nightly build.

**Comment by GregSutcliffe** [comment]
tests fine here. Merged, thanks @lzap !

**Pull request merged**